lib/spack/spack/binary_distribution.py

```diff
@@ -1111,6 +1117,11 @@ def relocate_package(spec, allow_root):
     # Hence 2 dictionaries are maintained here.
     prefix_to_prefix_text = OrderedDict({})
     prefix_to_prefix_bin = OrderedDict({})
+
+    if old_sbang_install_path:
+        import spack.hooks.sbang as sbang
+        prefix_to_prefix_text[old_sbang_install_path] = sbang.sbang_install_path()
```

Comment:
The sbang logic should go after the prefix logic in the ordered dict, otherwise we will cause issues by replacing the first part of the prefix (the subset that is the sbang prefix) before going to replace the prefix, and the prefixes won't match when we handle the full prefix.

Reply:
`sbang.sbang_install_path()` is the full path to sbang, so shouldn't it go before the prefix logic?

for instance:
```
config:
  install_tree:
    root: /spack-software
```

`sbang.sbang_install_path()` would return `/spack-software/bin/sbang`